packages/modals/src/elements/DrawerModal/DrawerModal.tsx

```diff
@@ -49,12 +58,32 @@ const DrawerModalComponent = forwardRef<HTMLDivElement, IDrawerModalProps>(
       useModal({
         idPrefix: id,
         modalRef,
-        focusOnMount,
-        restoreFocus,
+        focusOnMount: false /* [1:b] */,
+        restoreFocus: false /* [1:b] */,
         environment,
         onClose
       });
 
+    /* [1:c] */
+    useEffect(() => {
+      if (environment) {
+        if (modalRef.current && isOpen) {
+          if (restoreFocus === undefined ? true : restoreFocus) {
+            triggerRef.current = activeElement(environment) as HTMLElement;
+          }
+
+          if (focusOnMount === undefined ? true : focusOnMount) {
+            modalRef.current.focus();
+          }
+        }
+
+        if (triggerRef.current && !isOpen) {
+          triggerRef.current.focus();
+          triggerRef.current = null;
+        }
+      }
+    }, [environment, modalRef, triggerRef, restoreFocus, focusOnMount, isOpen]);
```

Comment:
The implementation is intentionally verbose for the sake of readability. If we find this unnecessary, we can compact this further.